### ISSUE TITLE
Monitor that apache logs on simian are getting updated regularly

### ISSUE THREAD

**Issue by mattpen** [phetsims/website#522]
We should add a script or monitor on simian that ensures that the Apache httpd log files copied from the production servers are always being updated.  We could potentially add a startup script, or perhaps Nagios can be customized for this purpose.

@markmorlino - Do you have any suggestions on how to monitor the logs?  We want to make sure that there is are frequent updates, anywhere from every 5 seconds to 5 minutes could be acceptable.

**Comment by markmorlino** [comment]
We can monitor the log files on simian. I've had to do something very similar before (for slightly different reasons) and it ends up being a very simple shell script. The problem I see with this method is that by the time it alerts us, we would most likely be losing log information already. Also, In the past, doing this has resulted in false alerts but that was for a site that really only had traffic from the US so there were times at night when there would be longer gaps between requests. 

I think it's fairly safe to assume that apache will log on the cache servers as long as it is running and there is free disk space. And while I don't particularly care for the syslog forwarding as a transfer mechanism, it has not failed to work yet. Every time we've lost log data like this it has been because apache was no longer able to write to a full disk on the cache servers.

I can't find the email but I think OIT had told us that they were monitoring the disks on the cache servers with nagios already. I think we should ask them to alert us when /data gets to be 80-90% full. I also think we should increase the size of /data such that it can hold an additional few days of data more than we expect it to. They seem to log less than 2g per day per cache server. If we made /data 20g then we would have room for 2-3 days of logs left when it became 80% full. This would give us some time to deal with the situation before we started losing log data.

For now I will write a script to tell us if the files on simain have not been written to in the past 5 minutes and cron it to run every five minutes. This shouldn't take very long to setup.

At some point we should probably chat about what to do moving forward with the simian replacement. Unless there is a need for near-realtime log data on simian, I think we might be better off not using syslog. I think using rsync to copy rotated logs off of the cache servers as a logrotate post-rotate script would be cleaner. This would mean that logs would get copied to simain once per day.

TLDR:
- I'll increase /data on phet-cache1/2 to 20g
- I'll write a monitor script to watch the files on simian
- Let's ask OIT to have nagios alert us when /data is over 80% full (can you see if you have the email from them about nagios?)
- Let's talk about re-visiting this process when setting up the replacement server, if transferring the files once per day is an option.

**Comment by mattpen** [comment]
>I'll increase /data on phet-cache1/2 to 20g
>I'll write a monitor script to watch the files on simian

This sounds great, thanks!

> Let's ask OIT to have nagios alert us when /data is over 80% full (can you see if you have the email from them about nagios?)

@markmorlino We have disk space alerts setup with NewRelic, would this work just as well?  I added you as a user, the url is https://rpm.newrelic.com/accounts/649073/servers.  If you go to the [server alerts policy](https://rpm.newrelic.com/accounts/649073/server_alert_policies) and click edit, you can change your email preferences for the alert.

**Comment by markmorlino** [comment]
@mattpen thanks, NewRelic should work just fine. I've got that script running in my home directory and emailing me. Once I'm satisfied that it isn't going to turn into a spambot I will move it to a different directory and add you and @jonathanolson to the recipient list
